build-support/src/main/kotlin/com/squareup/wire/buildsupport/WireBuildPlugin.kt

```diff
@@ -196,22 +198,26 @@ class WireBuildPlugin : Plugin<Project> {
   }
 
   private fun Project.configureCommonKotlin() {
-    tasks.withType(KotlinCompile::class.java).configureEach {
-      kotlinOptions {
-        freeCompilerArgs += listOf(
-          // https://kotlinlang.org/docs/whatsnew13.html#progressive-mode
-          "-progressive",
-          "-Xexpect-actual-classes",
+    tasks.withType(KotlinCompilationTask::class.java).configureEach {
+      compilerOptions {
+        languageVersion.set(KotlinVersion.KOTLIN_2_0)
+        freeCompilerArgs.addAll(
+          listOf(
+            "-Xexpect-actual-classes",
+          ),
         )
       }
     }
 
     val javaVersion = JavaVersion.VERSION_1_8
     tasks.withType(KotlinJvmCompile::class.java).configureEach {
-      kotlinOptions {
-        jvmTarget = javaVersion.toString()
-        freeCompilerArgs += listOf(
-          "-Xjvm-default=all",
+      compilerOptions {
+        languageVersion.set(KotlinVersion.KOTLIN_2_0)
+        jvmTarget.set(JvmTarget.fromTarget(javaVersion.toString()))
```

Comment:
We probably want to set `-Xjdk-release` in addition to this. I'm going to tackle that across all projects at some point, so no need to do right now.